fulcio.proto

```diff
@@ -0,0 +1,120 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.fulcio.v1;
+
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/empty.proto";
+
+option go_package = "github.com/sigstore/fulcio/pkg/generated/protobuf";
+option java_package = "dev.sigstore.fulcio.v1";
+option java_multiple_files = true;
+option java_outer_classname = "FulcioProto";
+
+service CA {
+    /**
+     * Returns information about the current state of the transparency log
+     */
+    rpc CreateSigningCertificate(CreateSigningCertificateRequest) returns (SigningCertificate){
+        option (google.api.http) = {
+          post: "/api/v2/signingCert"
+          body: "*"
+        };
+    }
+    /**
+     * Returns the bundle of certificates that can be used to validate the signed certificate timestamp
+     */
+    rpc GetTrustBundle(google.protobuf.Empty) returns (TrustBundle){
+        option (google.api.http) = {
+          get: "/api/v2/trustBundle"
+        };
+    }
+}
+
+message CreateSigningCertificateRequest {
+    /*
+     * Identity information about who possesses the private / public key pair presented
+     */
+    Credentials credentials    = 1 [(google.api.field_behavior) = REQUIRED];
+    /*
+     * The public key to be stored in the requested certificate
+     */
+    PublicKey public_key       = 2 [(google.api.field_behavior) = REQUIRED];
+    /*
+     * Proof that the client possesses the private key; must be verifiable by provided public key
+     */
+    bytes proof_of_possession  = 3 [(google.api.field_behavior) = REQUIRED];
```

Comment:
I like this name change from the generic "challenge". Its much more clear. Perhaps we could even explicitly note here that its the signature of the `sub` claim from the OIDC token? Note to self: we need to update the some docs to match I think once this gets merged.